torch_xla/csrc/ops/dynamic_ir.cpp

```diff
@@ -24,6 +25,17 @@ SizeNode::SizeNode(torch::lazy::Value input, size_t dim)
   upper_bound_ = xla_node->xla_shape(operand(0).index).dimensions(dim_);
 };
 
+int64_t SizeNode::getDynamicValue() const {
+  torch::lazy::NodePtr cloned =
+      torch::lazy::MakeNode<SizeNode>(operands_[0], dim_);
+  std::vector<XLATensorPtr> dummy_size_tensors = {
+      XLATensor::Create(cloned, *GetDefaultDevice(), at::ScalarType::Long)};
+  // TODO: cache the result
+  // TODO: add real dynamic test
+  std::vector<at::Tensor> res = XLATensor::GetTensors(&dummy_size_tensors);
+  return res[0].sum().item().toInt();
```

Comment:
do we really need to sum()? SizeNode is lowered to `GetDimensionSize` which should already be a 1-element tensor?

Reply:
good point, I will try to get rid of it.

Reply:
I see `sum()` being part of the return call 
How did this conversation evolve @JackCaoG?

Reply:
I think I removed it in another pr.